style/react/README.md

```diff
@@ -0,0 +1,50 @@
+# React
+
+ * Use React in [Strict Mode]
+ * Use React with [TypeScript](../typescript/README.md)
+ * Use [Jest](../testing-jest/README.md) for unit testing
+ * Use [styled-components] for CSS-in-JS
+ * Use [Apollo] with [code generation] when working with [GraphQL]
+ * Use [React Router] for routing in SPAs
+ * Avoid deeply nested routing when using React Router
+ * Prefer [Function Components] over [Class Components]
+ * Prefer keeping a single component in each file
+ * Use `PascalCase` for component names and their file names
+ * Use [React Hooks]
+ * Use pre-built hooks when possible (e.g. [streamich/react-use])
+ * Use [custom hooks] to encapsulate stateful logic outside a component
+ * Avoid [Forwarding Refs]
+ * Avoid [Higher-Order Components] and [recompose]
+ * Prefer the `children` prop over [render props]
+ * Prefer using [TypeScript prop interfaces] over [PropTypes]
+ * Prefer the [short syntax] when using [Fragments]
+ * Prefer [React Contexts] over [Redux]
+ * Prefer [Controlled Components] over [Uncontrolled Components]
+ * Avoid using indexes as the value for [keys]
```

Comment:
From the React docs (https://reactjs.org/docs/lists-and-keys.html#keys):

> When you don’t have stable IDs for rendered items, you may use the item index as a key as a last resort: <example>
> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.

Since React uses the keys to identify DOM elements, it won't re-render a DOM element if the key does not change. If you had a list of three items and you added a fourth and removed one at the same time, indexes would not uniquely identify the elements, which may lead to the change not triggering a re-render.

Reply:
Makes sense to me!